packages/components/src/slot-fill/bubbles-virtually/use-slot.ts

```diff
@@ -1,42 +1,17 @@
 /**
  * WordPress dependencies
  */
-import { useMemo, useContext } from '@wordpress/element';
+import { useContext } from '@wordpress/element';
 import { useObservableValue } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import SlotFillContext from './slot-fill-context';
-import type {
-	SlotFillBubblesVirtuallyFillRef,
-	SlotFillBubblesVirtuallySlotRef,
-	FillProps,
-	SlotKey,
-} from '../types';
+import type { SlotKey } from '../types';
 
 export default function useSlot( name: SlotKey ) {
 	const registry = useContext( SlotFillContext );
 	const slot = useObservableValue( registry.slots, name );
-
-	const api = useMemo(
-		() => ( {
-			updateSlot: (
-				ref: SlotFillBubblesVirtuallySlotRef,
-				fillProps: FillProps
-			) => registry.updateSlot( name, ref, fillProps ),
-			unregisterSlot: ( ref: SlotFillBubblesVirtuallySlotRef ) =>
-				registry.unregisterSlot( name, ref ),
-			registerFill: ( ref: SlotFillBubblesVirtuallyFillRef ) =>
-				registry.registerFill( name, ref ),
-			unregisterFill: ( ref: SlotFillBubblesVirtuallyFillRef ) =>
-				registry.unregisterFill( name, ref ),
-		} ),
-		[ name, registry ]
-	);
-
-	return {
-		...slot,
-		...api,
-	};
+	return { ...slot };
```

Comment:
Returning `{ ...slot }` instead of `slot` has two purposes:
1. The hook always returns an object, even if the `slot` is `undefined`. Important for compatibility.
2. The hook has a nice return type: an object with optional `ref` and `fillProps` fields.